### PR TITLE
Clarify usage of --force when used on a swarm manager

### LINKED ISSUE #26125: Clarify --force in swarm leave documentation
The CLI reference docs say this about `swarm leave --force`.

```
      --force   Force leave ignoring warnings.
```

They should also mention that `swarm leave --force` allows you to remove a manager, but does not gracefully remove the manager from the swarm. Doing this may cause the swarm to lose its quorum. Unless you don't care about the impact on the existing cluster, you should always demote managers to workers before removing them.

cc @sfsmithcha

### PR DESCRIPTION
**Pull request by mdlinville**
**\- What I did**
Documented caveats about using --force with a manager, and also showed how to remove a node from the manager to augment the existing example of removing the node from the node's CLI.

**\- Description for the changelog**
Clarify the behavior of using --force when removing a manager from a swarm

Fixes #26125

Signed-off-by: Misty Stanley-Jones misty@docker.com

### CODE REVIEW

File: docs/reference/commandline/swarm_leave.md
Comment (sfsmithcha) [inline docs/reference/commandline/swarm_leave.md]: The swarm can lose quorum if you demote a manager to worker, too. So it does not "prevent" losing quorum. To prevent that you need to a) join a new manager or b) promote a node to manager

**Comment by mstanleyjones** [comment]
@aaronlehmann PTAL

File: api/client/swarm/leave.go
Comment (aaronlehmann) [inline api/client/swarm/leave.go]: Maybe "Force this node...", since the command is always acting on the node where the command is run?

File: docs/reference/commandline/swarm_leave.md
Comment (aaronlehmann) [inline docs/reference/commandline/swarm_leave.md]: I suggest saying upfront that `swarm leave` won't allow a manager to leave unless `--force` is specified. This is implied by "To avoid needing to use `--force`...", but I think it would be clearer to say it directly.

How about changing

> When used on a manager, the swarm may lose quorum. To prevent this, join a new manager or promote an existing node to a manager

to

> The `--force` option will allow a manager to leave the swarm, but this does not reconfigure the swarm to cleanly remove the node as a manager. This can put the swarm in a bad state where expects more managers than the actual number, which can cause it to lose quorum. The safe way to remove a manager from a swarm is to demote it to a worker and then have it leave the swarm without `--force`. `--force` should only be used in situations where the swarm will no longer be used after the manager leaves, such as a single-node swarm.

File: docs/reference/commandline/swarm_leave.md
Comment (aaronlehmann) [inline docs/reference/commandline/swarm_leave.md]: I believe `node rm` can only be used on an active worker if you pass the `--force` argument.

**Comment by mstanleyjones** [comment]
bump

File: cli/command/swarm/leave.go
Comment (aaronlehmann) [inline cli/command/swarm/leave.go]: I'm a little concerned that this could be misleading, because it makes it sound like the command will automatically demote a worker. I think it's best to leave this as `Leave a swarm`, since other commands' short descriptions are short phrases.

File: docs/reference/commandline/swarm_leave.md
Comment (aaronlehmann) [inline docs/reference/commandline/swarm_leave.md]: I noticed that the node named `worker1` below is actually a manager (it has "manager status" listed as "Reachable"). Can you please remove the "Reachable" to make its appearance consistent with a worker?

File: docs/reference/commandline/swarm_leave.md
Comment (aaronlehmann) [inline docs/reference/commandline/swarm_leave.md]: "To remove an inactive node from the node list, use `docker node rm` on a manager node."

I'm not sure we should talk about `rm`'s `--force` flag here. That's a way to kick a node out of a swarm, from a manager. Generally users would want to use `docker swarm leave` instead.

This whole section should be moved underneath the "On a manager node:" example below, which shows output that would be changed by running `docker node rm worker2`.

**Comment by mstanleyjones** [comment]
Addressed @aaronlehmann latest comments.

File: docs/reference/commandline/node_rm.md
Comment (aaronlehmann) [inline docs/reference/commandline/node_rm.md]: Replace "master" with "manager"

File: docs/reference/commandline/swarm_leave.md
Comment (aaronlehmann) [inline docs/reference/commandline/swarm_leave.md]: Extra spae after `However,`

File: cli/command/swarm/leave.go
Comment (aaronlehmann) [inline cli/command/swarm/leave.go]: I'd prefer to make this more succinct. For context, here's the relevant part of the output of `docker swarm --help` in 1.12:

```
Commands:
  init        Initialize a swarm
  join        Join a swarm as a node and/or manager
  join-token  Manage join tokens
  update      Update the swarm
  leave       Leave a swarm
```

I think `Leave the swarm` is fine, but if you feel strongly about noting that it can only be run from a worker, how about `Leave the swarm (workers only)`?

Comment (aaronlehmann) [inline cli/command/swarm/leave.go]: Also, there should be no period at the end for consistency with the other lines.

File: cli/command/swarm/leave.go
Comment (aaronlehmann) [inline cli/command/swarm/leave.go]: I don't think there should be a period here either (preexisting problem, though).

File: docs/reference/commandline/node_rm.md
Comment (thaJeztah) [inline docs/reference/commandline/node_rm.md]: nit: stray space before comma

**Review by thaJeztah — COMMENTED** [review]
Only a minor nit, no show-stopper, so LGTM unless @aaronlehmann had more suggestions

**Comment by thaJeztah** [comment]
ping @aaronlehmann PTAL; I think it's ready to be merged

**Comment by aaronlehmann** [comment]
LGTM. Please address @thaJeztah's comment before merging though.

**Comment by mstanleyjones** [comment]
Fixed the nit. This is now ready to merge, @thaJeztah

**Review by thaJeztah — APPROVED** [review]
perfect! LGTM

**Comment by thaJeztah** [comment]
@mstanleyjones since this is both _code_ and _docs_ changes; perhaps I should remove the docs-cherry-pick on this one? (it will then go out with the 1.13 release), or do you want to cherry-pick just the docs changes?

**Comment by mstanleyjones** [comment]
The doc changes should not be cherry-picked independent of the code changes (it's to keep the usage messages in sync with the docs). I think this would be safe to cherry-pick back to 1.12 but that might be @aaronlehmann call. Would changing the output of -h break the contracts for backward compatibility?

**Comment by aaronlehmann** [comment]
Seems okay to me to change the help text in 1.12.

**Comment by thaJeztah** [comment]
Ok, adding to the milestone for consideration

**Comment by thaJeztah** [comment]
all 💚, merging

**Pull request merged**
